ui/src/actions/user.js

```diff
@@ -53,48 +53,50 @@ export const fetchUser = () => async (dispatch, getState) => {
   }
 };
 
-export const changeUsername = (newUsername) => async (dispatch, getState) => {
-  dispatch({ type: CHANGE_USERNAME_START });
+export const changeUsername =
+  (user, newUsername) => async (dispatch, getState) => {
+    dispatch({ type: CHANGE_USERNAME_START });
+
+    const token = getState().auth.token;
+
+    const config = {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+        authorization: token,
+      },
+      body: JSON.stringify({
+        new_username: newUsername,
+      }),
+    };
+
+    try {
+      const res = await fetch("/api/v1/auth/username", config);
```

Comment:
I don't think there is a need to fetch the username again if we make the request for changing a username return the updated username. At that point we need to simply dispatch it.

Reply:
I'm not sure I understand what you meant.
This is the PATCH request to change the username, it doesn't return the username.

Reply:
I feel like I'm missing something regarding how this should work in the frontend. (Why should the server return the username if I already have it?).
I'll remove this for now.

Reply:
That's a good point. I guess we can do that as well. Since we already have the username we can just set it after the patch request succeeds.

Reply:
Isn't that what I'm doing in the code?

Reply:
Yep, for some reason I misread the code and thought you were fetching the username again after the patch. My bad, slow mornings.

This LGTM but I will give this a spin locally just in case :).